docs/releasing.md

```diff
@@ -0,0 +1,32 @@
+# Release process
+
+## Manual steps
+
+### Expected artifacts
+
+1. A container image containing the aws-provider manager binary
+2. A release tarball containing the manifest-templates and a script to generate
+   the actual manifests
+3. Release notes
```

Comment:
Should we also list the actual github release here? We *could* also list the git tag here as well, but that can be implicitly implied from the github release.

Reply:
The tag imo is less important since the repo isn't designed to be imported from another library